yarn-project/stdlib/src/snapshots/download.ts

```diff
@@ -50,10 +57,66 @@ export function makeSnapshotPaths(baseDir: string): SnapshotDataUrls {
   return fromEntries(SnapshotDataKeys.map(key => [key, `${baseDir}/${key}.db`]));
 }
 
+function isGzipMagic(data: Buffer): boolean {
+  return data.length >= 2 && data[0] === 0x1f && data[1] === 0x8b;
+}
+
+function maybeGunzip(data: Buffer, contentEncodingHint?: string): Buffer {
+  const headerIndicatesGzip = contentEncodingHint?.toLowerCase().includes('gzip');
+  if (headerIndicatesGzip || isGzipMagic(data)) {
+    try {
+      const out = gunzipSync(data);
+      return out;
+    } catch {
+      // Fall through to return original if decompression fails
```

Comment:
Hey sorry for drive-by review - if we have a truncated gzip packet this may be a bit confusing as we'll just try it uncompressed, seems maybe better to just hard error?

Reply:
yeah I think that makes sense. But only if `isGzipMagic` is `true` I think as headers can lie